Units/parser-gdscript.r/sample.d/input.gd

```diff
@@ -0,0 +1,79 @@
+# Taken from https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_basics.html
+
+# A file is a class!
+
+# Inheritance
+
+extends BaseClass
+
+# (optional) class definition with a custom icon
+
+class_name MyClass, "res://path/to/optional/icon.svg"
```

Comment:
I wonder MyClass should be tagged with "class" kind?
How do you think?

Reply:
The class_name keyword in gdscript is a little bit weird. Actually, all files in GDScript are by default sort of anonymous classes. `extends` is class inheritance for files, and class_name sets a name for the class that the file defines. The identifier, which in this case is MyClass, is actually undefined inside the file where class_name is used. Instead, class_name automatically includes the file as a class in all other files in the project, in this case using `MyClass` as the identifier, without the need for any explicit `extends` or `preload`. If we actually try to use the MyClass identifier within this file though, Godot will throw an error and you probably don't ever need to try reference MyClass from inside MyClass anyways because you can just directly refer to other variables and functions in here.

This actually means that the parser I've submitted is slightly incomplete, because it also needs to crawl the rest of the directory for .gd files with `class_name` declarations to list out every class that will be available in a file because GDScript can include files without you saying that you wanted to import anything, but I thought this might be a little out of scope.